docker-compose.yml

```diff
@@ -0,0 +1,62 @@
+version: "3.8"
+
+services:
+
+  opendevin-build:
+    build:
+      dockerfile: ./Dockerfile
+    image: opendevin:latest
+    container_name: opendevin-build
+    restart: unless-stopped
+    command: make build
+    healthcheck:
+      test: pytest --version
+      interval: 5s
+      timeout: 2s
+      retries: 6000
+    networks:
+      - opendevin-network
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - ./:/OpenDevin
+      - opendevin-volume:/usr
+  
+  backend:
+    build: .
+    image: opendevin:latest
+    container_name: opendevin-backend
+    restart: unless-stopped
+    command: make start-backend
+    ports:
+      - 3000:3000
+    networks:
+      - opendevin-network
+    volumes_from:
+      - opendevin-build
+    depends_on:
+      opendevin-build:
+        condition: service_healthy
+
+  frontend:
+    build: .
```

Comment:
I wanted to keep fronted and backend as separate docker services (so that if some just want say backend running). It is not being built, the frontend and backend depend on the opendevin-build service onces its built sucessfully (line 32 and 56). Infact, we can also export this container so that people just download it, do docker load and do not have to even wait 5-10 mins it takes for all the dependencies to install via poetry!.. :)

Reply:
I'm not a fan of packing both frontend and backend into the same image if you're only going to run either of the two. Makes the image size bloated.

Here's a slightly different approach to build slimmer and separate images: https://github.com/OpenDevin/OpenDevin/pull/481/files

Reply:
Yeah, I saw... you are creating separate images for frontend and backend, I have made separate containers depending on the same image. what are the image size you are getting? mine is 1.29GB... anyhow the opendevin/sandbox is 560MB... nothing too big these days. I am sure the backend image will be large only... so many dependencies... :)... even the base node image is 1.1 GB... XO